### PR TITLE
Update to RKE 1.3.11

### PR DESCRIPTION
**Pull request by tsde**
Issue: https://github.com/rancher/terraform-provider-rke/issues/342

Bump RKE version to 1.3.11

This version uses etcd v3.5.3 that includes the fix for [this issue](https://github.com/etcd-io/etcd/issues/13766). Due to this etcd bug, versions 1.3.9 was not recommended for production use.

I'm far form being a golang guru so I based my changes mainly on @VIKMSTR and @fe-ax already opened PRs. Since 1.3.10 ships with the etcd fix, I think we should concentrate on this PRs if that's OK for you @VIKMSTR and @fe-ax. I successfully built the provider and used it to build a test cluster with 3 control planes nodes and 2 workers. Still, I may have missed some key changes that should be made, so double checking would be wise ^^.

Changes includes
- Update Go version to 1.17.9 (1.17 is the version used by RKE)
- Updating dependencies in `go.mod`. I updated all deps to their latest available version except for `github.com/hashicorp/terraform-plugin-sdk`. I didn't upgrade it to v2 because it seems like many changes in provider code should be made to make it compatible so this should be part of a dedicated PR IMO. Therefore, I simply updated this dep to the last v1.x available
- Force `github.com/spf13/afero` to stick to `v1.2.2` required by the version of `github.com/hashicorp/terraform-plugin-sdk` mentioned above

Can anyone from Rancher shed some lights on the current status of this project ? Activity seemed to have drastically dropped during the last few months. I can understand that efforts are now mainly directing towards Rancher 2 / RKE 2 products but for users of RKE 1, this provider is still very useful. A little bit of visibility would be great :wink:  Maybe @rawmind0 or @annablender can help with this PR ?

### CODE REVIEW

**Comment by VIKMSTR** [comment]
Thumbs up!
Make sense to merge this instead. 

I'm also kind of interested what's the status of the project, how it looks like with the future support and possibly with further support of RKE2.

**Review by VIKMSTR — APPROVED** [review]

**Comment by tsde** [comment]
I've just updated the PR to use RKE 1.3.11 as it was released a week ago. Kubernetes 1.23 is now officially supported.
It would be great to have some feedback and a review from Rancher team. @HarrisonWAffel @rawmind0 @annablender Any chance to see this PR merged soon ? Any blocker that would prevent this PR from being merged ?

**Comment by annablender** [comment]
I'll review but @kinarashah has more experience with upgrading RKE for Kubernetes 1.24, so I'll wait for her review on this one. DNM, pending 2.6.5 Terraform release.

**Comment by tsde** [comment]
Thanks for your feedback @annablender

File: go.mod
Comment (kinarashah) [inline go.mod:196]: nit: could all replace statements be placed under one replace directive?

**Review by kinarashah — COMMENTED** [review]
@annablender Changes LGTM, could you look at the test cases we'd need to verify? Looks like a few things have already been tested per the PR description but would be helpful to have a list of things to check for both dev and QA testing. 

@tsde We still support RKE1 and release new k8s updates, bug fixes etc for RKE1. I'll try to find more information about terraform's release cycle for RKE1. Thank you for bringing in 1.23 for terraform.

**Comment by annablender** [comment]
I've added a new issue for the RKE update to 1.3.11 https://github.com/rancher/terraform-provider-rke/issues/342. Test notes will be added there. Terraform has an RKE test suite that can be run using `make testacc` and it would be prudent to make sure enabling the dockerd CRI works on an RKE cluster as well.

**Review by annablender — APPROVED** [review]

File: go.mod
Comment (tsde) [reply go.mod]: Good catch, just fixed ^^

**Comment by tsde** [comment]
> @tsde We still support RKE1 and release new k8s updates, bug fixes etc for RKE1. I'll try to find more information about terraform's release cycle for RKE1. Thank you for bringing in 1.23 for terraform.
@kinarashah Thank you for your feedback on this matter :+1:

**Review by annablender — APPROVED** [review]

**Comment by annablender** [comment]
Terraform RKE [v1.3.1 patch](https://github.com/rancher/terraform-provider-rke/releases/tag/v1.3.1) is released. Ok to merge

**Review by jakefhyde — APPROVED** [review]

**Pull request merged**
